src/descriptors/element_clip_edge.js

```diff
@@ -0,0 +1,195 @@
+	var clipLeftPx = this.computeClipLeftPxWidth(matches[4]);
+
+	var domElement = this._element.toDomElement();
+	var boundingRect = domElement.getBoundingClientRect();
+
+	var clipRect = {
+		left: boundingRect.left + clipLeftPx,
+		right: boundingRect.left + clipRightPx,
+
+		top: boundingRect.top + clipTopPx,
+		bottom: boundingRect.top + clipBottomPx
+	};
+
+	clipRect.width = clipRect.right - clipRect.left;
+	clipRect.height = clipRect.bottom - clipRect.top;
+
+	return clipRect;
+};
+
+var LENGTH_EXPR_PATTERN = /([\-\+]?[0-9\.]+)([a-zA-Z]+)/;
+
+ElementClipEdge.prototype.computeCssPxForLengthInElement = function computeCssPxForLengthInElement(lengthExpr) {
+	var matches = lengthExpr.match(LENGTH_EXPR_PATTERN);
+
+	if(!matches) {
+		ensure.unreachable("CSS length expression expected, got " + lengthExpr);
+	}
+
+	// convert the parsed number part of the lengthExpr to px by multiplying it by the computed ratio of lengthExpr's
+	// css unit to css px units.
+	return parseFloat(matches[1]) * this.computeCssUnitToCssPxRatio(matches[2]);
+};
+
+ElementClipEdge.prototype.computeClipTopPxHeight = function computeClipTopPxHeight(lengthExpr) {
+	return this.computeCssPxForLengthInElement((lengthExpr === "auto") ? "0px" : lengthExpr);
+};
+
+ElementClipEdge.prototype.computeClipRightPxWidth = function computeClipRightPxWidth(lengthExpr) {
+	if(lengthExpr === "auto") {
+		// "auto" for clip rect's right component will be the width of the element, enclosing the borders but not the
+		// margins.  offsetWidth gives us this value
+
+		return this._element.toDomElement().offsetWidth;
+	}
+
+	return this.computeCssPxForLengthInElement(lengthExpr);
+};
+
+ElementClipEdge.prototype.computeClipBottomPxHeight = function computeClipBottomPxHeight(lengthExpr) {
+	if(lengthExpr === "auto") {
+		// "auto" for clip rect's bottom component will be the height of the element, enclosing the borders but not the
+		// margins.  offsetHeight gives us this value
+
+		return this._element.toDomElement().offsetHeight;
+	}
+
+	return this.computeCssPxForLengthInElement(lengthExpr);
+};
+
+ElementClipEdge.prototype.computeClipLeftPxWidth = function computeClipLeftPxWidth(lengthExpr) {
+	return this.computeCssPxForLengthInElement((lengthExpr === "auto") ? "0px" : lengthExpr);
+};
+
+var PX_TEST_CSS_STYLES = "padding: 0; margin: 0; border: 0; visibility: hidden; position: absolute; height: 0";
+
+ElementClipEdge.prototype.computeCssUnitToCssPxRatio = function computeCssUnitToCssPxRatio(unit) {
+	var domElement = this._element.toDomElement(),
+		parentElement = domElement.parentNode;
+
+	if(unit !== "px" && parentElement) {
+		var testEl = domElement.ownerDocument.createElement("div");
+		testEl.setAttribute("style", PX_TEST_CSS_STYLES + "; width: 100" + unit + ";");
+		parentElement.appendChild(testEl);
+		var ratio = testEl.offsetWidth / 100;
+		parentElement.removeChild(testEl);
+		return ratio;
+	}
+
+	return 1;
+};
+
+// default module export
+module.exports = ElementClipEdge;
```

Comment:
Style: declare the export along with the function, and use `Me` for the current class. So line 16 should look like this:

```javascript
var Me = module.exports = function ElementClipEdge...
```